pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/LiteralOperand.java

```diff
@@ -28,7 +29,22 @@ public class LiteralOperand implements TransformOperand {
 
   public LiteralOperand(RexExpression.Literal rexExpression) {
     _resultType = rexExpression.getDataType();
-    _value = rexExpression.getValue();
+    _value = convert(rexExpression.getValue());
```

Comment:
We want to convert the value to the internal type. I think `TIMESTAMP` might not be handled correctly right now, where the raw value is not really the external value format `Timestamp`, so we may add a special case for that and add a TODO to fix that in the future
```suggestion
    _value = _resultType.toInternal(rexExpression.getValue());
```

Reply:
+1. this and the rest of the places, ProtoSer/De only provide an identical mapping from ser/de perspective. for this particular situation, we need to leverage the RexExpression.Literal structure, which not only encode the value but also the type

we also need to make sure that proto ser/de only accepts particular type -->
if the data is `ByteArray`, `byte[]`, or `ByteString` which can all ser/de into `ByteString` we should not allow that --> e.g. need to make sure the Ser/De is symmetrics

particularly in this case we should not allow the object to be GeogeorianCalendar --> it shouldl explicitly converted into long before setting it into the Literal

Reply:
Changed the proto serde to generate internal data type directly